packages/commerce-sdk-react/src/auth/index.ts

```diff
@@ -1164,55 +1190,66 @@ class Auth {
 
     /**
      * A wrapper method for commerce-sdk-isomorphic helper: authorizeIDP.
+     * Initiates OAuth2 authorization flow for Identity Provider (IDP) login.
      *
      */
-    async authorizeIDP(parameters: AuthorizeIDPParams) {
-        const redirectURI = parameters.redirectURI || this.redirectURI
+    async authorizeIDP(parameters: AuthorizeIDPPublicParams) {
+        const slasClient = this.client
         const usid = this.get('usid')
-        const customParameters = extractCustomParameters(parameters)
-        const {url, codeVerifier} = await helpers.authorizeIDP(
-            this.client,
-            {
-                redirectURI,
-                hint: parameters.hint,
-                ...(usid && {usid}),
-                ...customParameters
-            },
-            this.isPrivate
-        )
+        const dntPref = this.getDnt({includeDefaults: true})
+
+        // Extract known parameters and get custom ones
+        const {redirectURI, hint, ...customParams} = parameters
+        const finalRedirectURI = redirectURI || this.redirectURI
+
+        const authorizeParams = {
+            ...customParams,
+            ...(usid && {usid}),
+            ...(dntPref && {dnt: dntPref}),
+            redirectURI: finalRedirectURI,
+            hint: hint || ''
+        }
+
+        const result = await helpers.authorizeIDP({
+            slasClient,
+            parameters: authorizeParams
+        })
+
+        // Store code verifier for later use in loginIDPUser
+        this.set('code_verifier', result.codeVerifier)
 
         if (onClient()) {
-            window.location.assign(url)
+            window.location.assign(result.url)
         } else {
             console.warn('Something went wrong, this client side method is invoked on the server.')
         }
-        this.set('code_verifier', codeVerifier)
+
+        return result
     }
 
     /**
      * A wrapper method for commerce-sdk-isomorphic helper: loginIDPUser.
      *
      */
-    async loginIDPUser(parameters: LoginIDPUserParams) {
+    async loginIDPUser(parameters: LoginIDPUserPublicParams) {
```

Comment:
Actually, can we reuse the old name ie. `LoginIDPUserParams`. I'm not sure we need to have `Public` in the name for these